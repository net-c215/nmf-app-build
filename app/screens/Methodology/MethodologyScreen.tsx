import React from "react";
import { ScrollView, Linking, GestureResponderEvent } from "react-native";
import HTML from "react-native-render-html";

import styles from "./MethodologyScreen.styles";
import navigationOptions from "./Methodology.navigationOptions";
import methodology from "../../../assets/methodology/methodology.json";

const MethodologyScreen = () => {
  return (
    <ScrollView style={styles.container}>
      <HTML
        html={methodology[0].body}
        onLinkPress={(_: GestureResponderEvent, link: string) => {
          Linking.openURL(link);
        }}
        baseFontStyle={{ fontSize: 18 }}
      />
    </ScrollView>
  );
};

MethodologyScreen.navigationOptions = navigationOptions;

export default MethodologyScreen;
