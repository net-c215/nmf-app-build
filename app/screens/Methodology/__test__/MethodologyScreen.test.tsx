import React from "react";
import renderer from "react-test-renderer";
import MethodologyScreen from "../MethodologyScreen";

it("MethodologyScreen renders correctly", () => {
  const tree = renderer.create(<MethodologyScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
