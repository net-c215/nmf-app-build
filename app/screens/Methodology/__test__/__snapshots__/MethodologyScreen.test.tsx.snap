// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MethodologyScreen renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "paddingHorizontal": 14,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "marginTop": 22,
        }
      }
    >
      <Text.Primary
        style={
          Object {
            "paddingVertical": 10,
            "textAlign": "center",
          }
        }
      >
        Some Random Text
      </Text.Primary>
    </View>
  </View>
</RCTScrollView>
`;
